### PR TITLE
Implement additional feature for stopping and renaming the unstable VMs in case of failure

### PR DESCRIPTION
**Pull request by andy-mishechkin**
# Description
Improvement - move-vm.ps1 will stop  and rename unstable VMs in case of failure before moving.
Test failure: https://github.visualstudio.com/virtual-environments/_build/results?buildId=85939&view=logs&j=80d86624-2c9a-5cad-9434-144b838e69fb&t=027aaa20-7755-5366-f0d4-a9dbdf9aa415

#### Related issue:
https://github.com/actions/virtual-environments-internal/issues/1335

## Check list
- [X] Related issue / work item is attached
- [X] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images.CI/macos/move-vm.ps1 [hunk 1]
```diff
@@ -48,12 +48,23 @@ Import-Module $PSScriptRoot\helpers.psm1 -DisableNameChecking
 # Connection to a vCenter Server system
 Connect-VCServer
 
-try
-{
-    Get-VM $VMName | Move-VM -Datastore $TargetDataStore -ErrorAction Stop
-    Write-Host "VM has been moved successfully to target datastore '$TargetDataStore'"
+$Vm = Get-VM $VMName
```
Comment (maxim-lobanov) [inline images.CI/macos/move-vm.ps1:51]: nitpick: use `lowerCamelCase` for local variables, `$vm`

File: images.CI/macos/move-vm.ps1 [hunk 2]
```diff
@@ -48,12 +48,23 @@ Import-Module $PSScriptRoot\helpers.psm1 -DisableNameChecking
...
+
+if($env:AGENT_JOBSTATUS -eq 'Failed') {
```
Comment (maxim-lobanov) [inline images.CI/macos/move-vm.ps1:53]: nitpick: styling
`if (` here and in other places

**Review by maxim-lobanov — APPROVED** [review]

File: images.CI/macos/move-vm.ps1 [hunk 3]
```diff
@@ -48,12 +48,23 @@ Import-Module $PSScriptRoot\helpers.psm1 -DisableNameChecking
...
+    try {
+        if($Vm.PowerState -ne "PoweredOff") {
+            Stop-Vm -Vm $Vm -Confirm:$false -ErrorAction Stop
```
Comment (al-cheb) [inline images.CI/macos/move-vm.ps1:56]: Stop-Vm -> Stop-VM

Comment (al-cheb) [inline images.CI/macos/move-vm.ps1:56]: -Vm -> -VM

File: images.CI/macos/move-vm.ps1 [hunk 4]
```diff
@@ -48,12 +48,23 @@ Import-Module $PSScriptRoot\helpers.psm1 -DisableNameChecking
...
+        }
+        Set-Vm -Vm $Vm -Name "${VMName}_failed" -Confirm:$false -ErrorAction Stop
```
Comment (al-cheb) [inline images.CI/macos/move-vm.ps1:58]: Set-Vm -Vm -> Set-VM -VM

File: images.CI/macos/move-vm.ps1 (hunk 1 above)
Comment (al-cheb) [inline images.CI/macos/move-vm.ps1:51]: $Vm -> $vm to consistent with other scripts

Comment (andy-mishechkin) [reply images.CI/macos/move-vm.ps1]: Done

File: images.CI/macos/move-vm.ps1 (hunk 2 above)
Comment (andy-mishechkin) [reply images.CI/macos/move-vm.ps1]: Done

File: images.CI/macos/move-vm.ps1 (hunk 3 above)
Comment (andy-mishechkin) [reply images.CI/macos/move-vm.ps1]: Changed

File: images.CI/macos/move-vm.ps1 (hunk 4 above)
Comment (andy-mishechkin) [reply images.CI/macos/move-vm.ps1]: Done

File: images.CI/macos/move-vm.ps1 (hunk 1 above)
Comment (andy-mishechkin) [reply images.CI/macos/move-vm.ps1]: changed

File: images.CI/macos/move-vm.ps1 (hunk 3 above)
Comment (andy-mishechkin) [reply images.CI/macos/move-vm.ps1]: Done

File: images.CI/macos/move-vm.ps1 [hunk 5]
```diff
@@ -23,7 +23,7 @@ vCenter password (Example "12345678")
 param(
     [Parameter(Mandatory)]
     [ValidateNotNullOrEmpty()]
-    [string]$VMName,
+    [string]$vMName,
```
Comment (vsafonkin) [inline images.CI/macos/move-vm.ps1:26]: Why is a lowercase letter used here?

Comment (andy-mishechkin) [reply images.CI/macos/move-vm.ps1:26]: Sorry, autoreplacement bug :)
Fixed

**Review by vsafonkin — APPROVED** [review]

**Review by AlenaSviridenko — APPROVED** [review]

**Review by al-cheb — APPROVED** [review]

**Review by maxim-lobanov — APPROVED** [review]

**Pull request merged**